lib/internal/streams/end-of-stream.js

```diff
@@ -255,6 +268,15 @@ function eos(stream, options, callback) {
   return cleanup;
 }
 
+function eosWeb(stream, opts, callback) {
+  PromisePrototypeThen(
+    stream[kState].streamClosed,
+    () => callback.call(stream),
+    (err) => callback.call(stream, err)
```

Comment:
You need to call the callback with a process.nextTick

Reply:
understood updating

Reply:
Done

Reply:
fixed this to properly use process.nextTick